daemon/src/payout_curve.rs

```diff
@@ -511,19 +512,53 @@ mod tests {
             Usd::new(dec!(3500.00)),
             Leverage::new(5).unwrap(),
             200,
+            0,
         )
         .unwrap();
 
         pretty_assertions::assert_eq!(actual_payouts, expected_payouts());
     }
 
+    #[test]
+    fn verify_effect_of_funding_fee() {
+        let funding_fee = 100;
+        let actual_payouts = calculate_payout_parameters(
+            Price::new(dec!(54000.00)).unwrap(),
+            Usd::new(dec!(3500.00)),
+            Leverage::new(5).unwrap(),
+            200,
+            funding_fee,
+        )
+        .unwrap();
+
+        let adjusted_payouts = expected_payouts()
+            .into_iter()
+            .map(
+                |PayoutParameter {
+                     left_bound,
+                     right_bound,
+                     long_amount,
+                     short_amount,
+                 }| PayoutParameter {
+                    left_bound,
+                    right_bound,
+                    long_amount: long_amount.wrapping_sub(funding_fee),
+                    short_amount: short_amount + funding_fee,
+                },
+            )
+            .collect::<Vec<_>>();
+
+        pretty_assertions::assert_eq!(actual_payouts, adjusted_payouts);
+    }
```

Comment:
Is there any way we can assert this differently? You are basically reimplementing what you did above, that is not exactly the best assertion :D 

What about checking:

- That the sum for each payout is the same as without the fee
- That the number of payouts didn't change if you use a fee
- That the short gets the fee on every payout (map both iterators to just the short amount, zip them, subtract one from the other, assert that it all items are equal to the funding fee)

Combination of 1 and 3 should give us good confidence.

Reply:
I agree that this assertion is poor. I'll work on it and I'll try to incorporate your ideas.